opcua-core/src/errors.rs

```diff
@@ -0,0 +1,12 @@
+use opcua_types::Variant;
+use thiserror::Error;
+
+/// Rust OpcUa specific errors
+#[allow(missing_docs)]
+#[derive(Error, Debug)]
+pub enum OpcUAError {
+    #[error("Received an expected variant type")]
+    UnExpectedVariantType(Variant),
```

Comment:
This should probably contain a `VariantScalarTypeId`, if I understand its purpose correctly. You can easily get that from a variant that isn't `Empty`.

Reply:
Maybe `Option<VariantScalarTypeId>` I guess, to contain the `Empty` case.

Reply:
done